src/modules/mojo.rs

```diff
@@ -0,0 +1,155 @@
+use super::{Context, Module, ModuleConfig};
+
+use crate::configs::mojo::MojoConfig;
+use crate::formatter::StringFormatter;
+
+/// Creates a module with the current Mojo version
+pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
+    let mut module = context.new_module("mojo");
+    let config = MojoConfig::try_load(module.config);
+
+    let is_mojo_project = context
+        .try_begin_scan()?
+        .set_files(&config.detect_files)
+        .set_extensions(&config.detect_extensions)
+        .set_folders(&config.detect_folders)
+        .is_match();
+
+    if !is_mojo_project {
+        return None;
+    }
+
+    let parsed = StringFormatter::new(config.format).and_then(|formatter| {
+        formatter
+            .map_meta(|variable, _| match variable {
+                "symbol" => Some(config.symbol),
+                _ => None,
+            })
+            .map_style(|variable| match variable {
+                "style" => Some(Ok(config.style)),
+                _ => None,
+            })
+            .map(|variable| match variable {
+                "version" => {
+                    let mojo_version_output = context.exec_cmd("mojo", &["--version"])?.stdout;
+                    let version_items = mojo_version_output.split(' ').collect::<Vec<&str>>();
+
+                    if version_items.len() <= 1 {
+                        return None;
+                    }
+
+                    if version_items.len() == 2 || !config.show_commit {
+                        return Some(Ok(version_items[1].trim().to_string()));
+                    }
+
+                    let mojo_version = version_items[1].trim();
+                    let mojo_hash = version_items[2].trim();
+
+                    Some(Ok(format!("{} {}", mojo_version, mojo_hash)))
```

Comment:
I think it might be better to have separate `$version` and `$hash` variables. To cache the output/parsed version you can use `once_cell::sync::Lazy`.

Reply:
Hey, thanks a lot for the suggestions. I implemented the hash as a variable and used the match pattern. Much better indeed!

However, I'm struggling to understand how `Lazy` will be beneficial here. Since the module function will need to be executed anyway when `parsed` is used, wouldn't the function inside `Lazy` be called the same amount of times with or without `Lazy`? I see only a delay in the function call

Reply:
The version is not necessarily included in the format string, and in that case getting the version can be skipped altogether.

Reply:
Oh well, I wouldn't expect someone to use this without showing the version. But since it's a possibility, I believe I have it implemented now 👍.